include/arch/x86/multiboot.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_X86_MULTIBOOT_H_
+#define ZEPHYR_INCLUDE_ARCH_X86_MULTIBOOT_H_
+
+#ifndef _ASMLANGUAGE
+
+/*
+ * Multiboot (version 1) boot information structure.
+ *
+ * Only fields/values of interest to Zephyr are enumerated: at
+ * present, that means only those pertaining to the framebuffer.
+ */
+
+struct x86_multiboot_info {
```

Comment:
FWIW: there's some code in x86_64 (arch/x86_64/core/xuk-stub32.c) to generate a multiboot header and read a few words from this struct that could usefully be unified with this if you have the time.  The header is trivial and probably not worth the effort, but this struct definition would be nice to use.

Reply:
Ah, indeed.  There's actually a lot of unification that we between the x86_64 and x86 trees we could probably get around to ... that's something I wanted to talk to you about..